applications/StructuralMechanicsApplication/custom_elements/timoshenko_beam_element_2D2N.cpp

```diff
@@ -0,0 +1,1053 @@
+    p_new_elem->SetData(this->GetData());
+    p_new_elem->Set(Flags(*this));
+
+    // Currently selected integration methods
+    p_new_elem->SetIntegrationMethod(mThisIntegrationMethod);
+
+    // The vector containing the constitutive laws
+    p_new_elem->SetConstitutiveLawVector(mConstitutiveLawVector);
+
+    return p_new_elem;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void TimoshenkoBeamElement2D2N::EquationIdVector(
+    EquationIdVectorType& rResult,
+    const ProcessInfo& rCurrentProcessInfo
+    ) const
+{
+    KRATOS_TRY;
+
+    const auto& r_geom = GetGeometry();
+    const SizeType number_of_nodes = r_geom.size();
+    const SizeType dofs_per_node = GetDoFsPerNode(); // u, v, theta
+
+    if (rResult.size() != dofs_per_node * number_of_nodes)
+        rResult.resize(dofs_per_node * number_of_nodes);
+
+
+    for (IndexType i = 0; i < number_of_nodes; ++i) {
+        const SizeType index = i * dofs_per_node;
+        rResult[index]     = r_geom[i].GetDof(DISPLACEMENT_X).EquationId();
+        rResult[index + 1] = r_geom[i].GetDof(DISPLACEMENT_Y).EquationId();
+        rResult[index + 2] = r_geom[i].GetDof(ROTATION_Z    ).EquationId();
+    }
+
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void TimoshenkoBeamElement2D2N::GetDofList(
+    DofsVectorType& rElementalDofList,
+    const ProcessInfo& rCurrentProcessInfo
+    ) const
+{
+    KRATOS_TRY;
+
+    const auto& r_geom = GetGeometry();
+    const SizeType number_of_nodes = r_geom.size();
+    const SizeType dofs_per_node = GetDoFsPerNode(); // u, v, theta
+    rElementalDofList.resize(dofs_per_node * number_of_nodes);
+
+    for (IndexType i = 0; i < number_of_nodes; ++i) {
+        const SizeType index = i * dofs_per_node;
+        rElementalDofList[index]     = r_geom[i].pGetDof(DISPLACEMENT_X);
+        rElementalDofList[index + 1] = r_geom[i].pGetDof(DISPLACEMENT_Y);
+        rElementalDofList[index + 2] = r_geom[i].pGetDof(ROTATION_Z    );
+    }
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+const double TimoshenkoBeamElement2D2N::CalculatePhi(ConstitutiveLaw::Parameters &rValues)
+{
+    const auto &r_material_properties = rValues.GetMaterialProperties();
+    const double E   = r_material_properties[YOUNG_MODULUS];
+    const double A   = r_material_properties[CROSS_AREA];
+    const double I   = r_material_properties[IZ];
+    const double k_s = r_material_properties.Has(SHEAR_CORRECTION_XY) ? r_material_properties[SHEAR_CORRECTION_XY] : 5.0 / 6.0; // We assume rectangular shape
```

Comment:
Usually we consider `AREA_EFFECTIVE_Y` and `AREA_EFFECTIVE_Z` instead of the k..., this originates an inconsistent behaviour between the CR beam and this beam...